usr.bin/sed/process.c

```diff
@@ -672,11 +671,23 @@ regexec_e(regex_t *preg, const char *string, int eflags, int nomatch,
 		defpreg = preg;
 
 	/* Set anchors */
+#ifndef REG_STARTEND /* in cross-build tools */
```

Comment:
This looks different from the wrapper in grep.c (the former does not have a loop). Would it make sense to pull this out into a shared helper?